### PR TITLE
update, greenci, skip test if dplyr not available

### LINKED ISSUE #221: CI green for https://github.com/pawelru/ie_status?tab=readme-ov-file
todo

- [ ] depdencies test https://github.com/insightsengineering/rlistings/actions/runs/10437587898/job/28903936851
- [ ] nosuggests https://github.com/insightsengineering/rlistings/actions/runs/10437587898/job/28903939567

### PR DESCRIPTION
**Pull request by shajoezhu**
close #221

### CODE REVIEW

File: tests/testthat/test-export.R [hunk 1]
```diff
@@ -8,6 +8,9 @@ testthat::test_that("Listing print correctly", {
 })
 
 testthat::test_that("key columns repeat with export_as_txt", {
+  require("dplyr", quietly = TRUE)
+  skip_if_not_installed("dplyr")
+
```
Comment (pawelru) [inline tests/testthat/test-export.R:13]: it's better to skip first and then do something if not skipped already

Comment (shajoezhu) [reply tests/testthat/test-export.R:13]:
```suggestion
  skip_if_not_installed("dplyr")
  require("dplyr", quietly = TRUE)
```

File: vignettes/col_formatting.Rmd [hunk 2]
```diff
@@ -14,6 +14,14 @@ knitr::opts_chunk$set(
...
   comment = "#>"
 )
+
+suggested_dependent_pkgs <- c("dplyr")
+knitr::opts_chunk$set(
+  eval = all(vapply(suggested_dependent_pkgs, (requireNamespace),
+    logical(1),
+    quietly = TRUE
+  ))
+)
```
Comment (pawelru) [inline vignettes/col_formatting.Rmd:24]: - why in brackets? `(requireNamespace)`
- let's try to merge it with the previous options setter

File: tests/testthat/test-paginate_listing.R [hunk 3]
```diff
@@ -1,4 +1,7 @@
 testthat::test_that("pagination works vertically", {
+  require("dplyr", quietly = TRUE)
+  skip_if_not_installed("dplyr")
```
Comment (shajoezhu) [inline tests/testthat/test-paginate_listing.R:3]:
```suggestion
  skip_if_not_installed("dplyr")
  require("dplyr", quietly = TRUE)
```

File: tests/testthat/test-paginate_listing.R [hunk 4]
```diff
@@ -22,6 +25,9 @@ testthat::test_that("pagination works vertically", {
 })
 
 testthat::test_that("horizontal pagination with 0 or 1 key column specified works correctly", {
+  require("dplyr", quietly = TRUE)
+  skip_if_not_installed("dplyr")
```
Comment (shajoezhu) [inline tests/testthat/test-paginate_listing.R:29]:
```suggestion
  skip_if_not_installed("dplyr")
  require("dplyr", quietly = TRUE)
```

File: tests/testthat/test-paginate_listing.R [hunk 5]
```diff
@@ -84,6 +90,9 @@ testthat::test_that("horizontal pagination with 0 or 1 key column specified work
 })
 
 testthat::test_that("listing works with no vertical pagination", {
+  require("dplyr", quietly = TRUE)
+  skip_if_not_installed("dplyr")
```
Comment (shajoezhu) [inline tests/testthat/test-paginate_listing.R:94]:
```suggestion
  skip_if_not_installed("dplyr")
  require("dplyr", quietly = TRUE)
```

File: tests/testthat/test-paginate_listing.R [hunk 6]
```diff
@@ -104,6 +113,9 @@ testthat::test_that("listing works with no vertical pagination", {
 })
 
 testthat::test_that("checking vertical pagination line calculation.", {
+  require("dplyr", quietly = TRUE)
+  skip_if_not_installed("dplyr")
```
Comment (shajoezhu) [inline tests/testthat/test-paginate_listing.R:117]:
```suggestion
  skip_if_not_installed("dplyr")
  require("dplyr", quietly = TRUE)
```

File: vignettes/col_formatting.Rmd [hunk 7]
```diff
@@ -55,6 +63,7 @@ library(rlistings)
 For this example, we will use the dummy ADAE dataset provided within the `formatters` package as our data frame, which consists of 48 columns of adverse event patient data, and one or more rows per patient. For the purpose of this example, we will subset the data and only use the first 15 records of the dataset. We will create some `NA` values in the data to showcase how `NA` values can be formatted, and sort the data by what will be our key columns.
 
 ```{r}
+require(dplyr)
```
Comment (pawelru) [inline vignettes/col_formatting.Rmd:66]: it seems that there is a dedicated library-calls chunk in lines 59-61 - let's add it there

Comment (pawelru) [reply vignettes/col_formatting.Rmd:63]: actually - do we use `dplyr` in that vignette at all? I cannot identify any dplyr call in here. As opposed to the `vignettes/ref_footnotes.Rmd` where we use it.

If it's not used then let's not introduce conditional eval for simplicity

Comment (shajoezhu) [reply vignettes/col_formatting.Rmd]: arrange

Comment (shajoezhu) [reply vignettes/col_formatting.Rmd]: mutate

Comment (pawelru) [reply vignettes/col_formatting.Rmd]: ok

File: vignettes/col_formatting.Rmd [hunk 8]
```diff
@@ -10,9 +10,14 @@ vignette: >
...
 ```{r, include = FALSE}
+suggested_dependent_pkgs <- c("dplyr")
 knitr::opts_chunk$set(
   collapse = TRUE,
-  comment = "#>"
+  comment = "#>",
+  eval = all(vapply(suggested_dependent_pkgs, requireNamespace,
+    logical(1),
+    quietly = TRUE
+  ))
```
Comment (pawelru) [inline vignettes/col_formatting.Rmd:20]:
```suggestion
  eval = all(vapply(
    suggested_dependent_pkgs, 
    requireNamespace,
    logical(1),
    quietly = TRUE
  ))
```

File: vignettes/ref_footnotes.Rmd [hunk 9]
```diff
@@ -10,9 +10,14 @@ vignette: >
...
 ```{r, include = FALSE}
+suggested_dependent_pkgs <- c("dplyr")
 knitr::opts_chunk$set(
   collapse = TRUE,
-  comment = "#>"
+  comment = "#>",
+  eval = all(vapply(suggested_dependent_pkgs, requireNamespace,
+    logical(1),
+    quietly = TRUE
+  ))
```
Comment (pawelru) [inline vignettes/ref_footnotes.Rmd:20]:
```suggestion
  eval = all(vapply(
    suggested_dependent_pkgs, 
    requireNamespace,
    logical(1),
    quietly = TRUE
  ))
```

File: vignettes/ref_footnotes.Rmd [hunk 10]
```diff
@@ -32,10 +39,12 @@ When creating a listing with the `rlistings` package, you may want to add refere
 
 To demonstrate, we will create a basic listing below.
 
-We begin by loading in the `rlistings` package. 
+We begin by loading in the `rlistings` package.
 
 ```{r, message=FALSE}
 library(rlistings)
+require(dplyr)
+
```
Comment (pawelru) [inline vignettes/ref_footnotes.Rmd:47]:
```suggestion
library(dplyr)
```

File: vignettes/col_formatting.Rmd [hunk 11]
```diff
@@ -10,9 +10,16 @@ vignette: >
 ---
 
 ```{r, include = FALSE}
+suggested_dependent_pkgs <- c("dplyr")
 knitr::opts_chunk$set(
   collapse = TRUE,
-  comment = "#>"
+  comment = "#>",
+  eval = all(vapply(
+    suggested_dependent_pkgs,
```
Comment (pawelru) [inline vignettes/col_formatting.Rmd:18]:
```suggestion
    suggested_dependent_pkgs,
```

File: vignettes/ref_footnotes.Rmd [hunk 12]
```diff
@@ -10,9 +10,16 @@ vignette: >
 ---
 
 ```{r, include = FALSE}
+suggested_dependent_pkgs <- c("dplyr")
 knitr::opts_chunk$set(
   collapse = TRUE,
-  comment = "#>"
+  comment = "#>",
+  eval = all(vapply(
+    suggested_dependent_pkgs,
```
Comment (pawelru) [inline vignettes/ref_footnotes.Rmd:18]:
```suggestion
    suggested_dependent_pkgs,
```

**Review by pawelru — APPROVED** [review]
LGTM 👍

**Pull request merged**